### PR TITLE
Make .search() not fail silently without searchParams

### PR DESCRIPTION
**Pull request by firechicago**
Currently, if you call `.search()` with a `resourceType` but no `searchParams`, the method fails silently, returning `undefined`.

This PR makes two changes:
1) `.search({resourceType:'someResource')` is now valid (resolving to `base-url/someResource`).
2) If `.search()` is called without either a `resourceType` or `searchParams` it will throw an error, rather than failing silently.

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -572,11 +572,14 @@ class Client {
   search({ resourceType, compartment, searchParams, headers } = {}) {
     if (compartment && resourceType) {
       return this.compartmentSearch({ resourceType, compartment, searchParams, headers });
-    } else if (resourceType && searchParams) {
-      return this.resourceSearch({ resourceType, searchParams, headers });
+    } else if (resourceType) {
```
Comment (Jammjammjamm) [inline lib/client.js]: This doesn't seem right. I think `searchParams` are only optional for a compartment search.

File: lib/client.js [hunk 2]
```diff
@@ -572,11 +572,14 @@ class Client {
...
+      const params = searchParams || {};
+      return this.resourceSearch({ resourceType, params, headers });
     } else if (searchParams) {
       return this.systemSearch({ searchParams, headers });
     }
+
+    throw new Error('search requires either searchParams or a resourceType');
```
Comment (Jammjammjamm) [inline lib/client.js]: A test should be added for this error, and the jsdoc for this function should be updated with a `@throws` description.

I don't think the error text is quite right. I think the requirements are:
```
(compartment && resourceType) || searchParams
```
Looking at this made me realize that we don't support a compartment search without a `resourceType`, but I'll make a separate issue for that.

File: lib/client.js (hunk 1 above)
Comment (firechicago) [reply lib/client.js]: I guess I'm confused then. What would I do if I wanted to use fhir-kit-client to hit `/Patient` with no query params? Currently you're required to pass an empty `searchParams` which seems like an odd design decision. (Or should I be using some other method?)

Comment (Jammjammjamm) [reply lib/client.js]: Is that a valid fhir request?

Comment (firechicago) [reply lib/client.js]: It's worked on every FHIR dev server that I've played with. I'm not sure how else you would semantically ask for "Give me every Patient that you've got/I'm allowed to access"

Comment (firechicago) [reply lib/client.js]: Also https://www.hl7.org/fhir/http.html#search makes it look like the parameters are optional

**Comment by firechicago** [comment]
👍 Looks Good

**Pull request merged**